### ISSUE TITLE
Error not being reported when create is required with empty dict in snapshot

### ISSUE THREAD

**Issue by calumy** [15r10nk/inline-snapshot#70]
I would like to report a potential error case that I don't believe is being handled correctly. 

When a test of this form is run:
```python
def test_empty_snapshot() -> None:
    assert 1 == snapshot()
```

Test passes, but an error is produced during tear down:

```bash
========================================================================================================================= ERRORS ==========================================================================================================================
________________________________________________________________________________________________________ ERROR at teardown of test_empty_snapshot _________________________________________________________________________________________________________
your snapshot is missing a value run pytest with --inline-snapshot=create to create the value
===================================================================================================================== inline snapshot =====================================================================================================================
Error: 1 snapshots are missing values (--inline-snapshot=create)
================================================================================================================= short test summary info =================================================================================================================
ERROR tests/test_empty_snapshot_dict.py::test_empty_snapshot - Failed: your snapshot is missing a value run pytest with --inline-snapshot=create to create the value
=============================================================================================================== 1 passed, 1 error in 0.09s ================================================================================================================
```

However, if the following is run:
```python
def test_empty_snapshot() -> None:
    assert 1 == snapshot({})["key"]
```

The test passes, errors are returned but the test suite does not fail:
```bash
===================================================================================================================== inline snapshot =====================================================================================================================
Error: 1 snapshots are missing values (--inline-snapshot=create)

==================================================================================================================== 1 passed in 0.07s ====================================================================================================================
```

I would expect the behaviour from the first case (teardown error) to also be present on the second case.

**Comment by 15r10nk** [comment]
You are correct, this is an error. I will fix it in the next release.

Thank you for the report.

**Comment by 15r10nk** [comment]
fixed in 0.8.2
